eclair-core/src/main/scala/fr/acinq/eclair/json/JsonSerializers.scala

```diff
@@ -402,6 +404,11 @@ object GlobalBalanceSerializer extends MinimalSerializer({
     JObject(JField("total", JDecimal(o.total.toDouble))) merge Extraction.decompose(o)(formats)
 })
 
+private[json] case class MessageReceivedJson(pathId: Option[ByteVector], replyPath: Option[BlindedRoute], unknownTlvs: Seq[GenericTlv])
```

Comment:
It's a nit, but I'm mentioning it now because serialization is difficult to change afterwards since it would break backwards compatibility: what about using a map for the unknown tlvs?

```suggestion
private[json] case class MessageReceivedJson(pathId: Option[ByteVector], replyPath: Option[BlindedRoute], unknownTlvs: Map[UInt64, ByteVector])
```

It feels slightly better for applications using it, they will usually be interested in reading the contents of a specific custom tlv tag.

Reply:
Done